core/src/main/java/org/apache/druid/java/util/common/guava/LimitedSequence.java

```diff
@@ -52,6 +52,7 @@
     return new LimitedYielder<>(subYielder, limitedAccumulator);
   }
 
+  @SuppressWarnings("InnerClassMayBeStatic") // False alarm: flagged by IDEA inspections, but can't actually be static.
```

Comment:
Would be nice to describe the explanation in your PR in here so someone in the future can remove the suppression if the inner class it depends on becomes static

Reply:
I'll merge this and add the comment in a follow-up, just in the interests of time since the builds are broken. Good idea though.

Reply:
Here's the followup: https://github.com/apache/druid/pull/9298

Reply:
^ Sorry I should have mentioned that was optional. Thanks for fixing the build!